internal/cmd/skupper/site/kube/site_generate.go

```diff
@@ -38,11 +37,9 @@ func NewCmdSiteGenerate() *CmdSiteGenerate {
 }
 
 func (cmd *CmdSiteGenerate) NewClient(cobraCommand *cobra.Command, args []string) {
-	cli, err := client.NewClient(cobraCommand.Flag("namespace").Value.String(), cobraCommand.Flag("context").Value.String(), cobraCommand.Flag("kubeconfig").Value.String())
+	cli, err := client.NewGenerateCmdClient(cobraCommand.Flag("namespace").Value.String())
```

Comment:
This is not necessary, I suggest to just keep
cmd.namespace = cli.Namespace in the NewClient method.

If generate methods are not checking with a client it is fine to just not call them.

Reply:
removed function and only set namespace in NewClient